cephfs/file.go

```diff
@@ -255,3 +255,38 @@ func (f *File) Fallocate(mode FallocFlags, offset, length int64) error {
 	ret := C.ceph_fallocate(f.mount.mount, f.fd, C.int(mode), C.int64_t(offset), C.int64_t(length))
 	return getError(ret)
 }
+
+const (
+	// LockSH places a shared lock.
+	// More than one process may hold a shared lock for a given file at a given time.
+	LockSH = int(C.LOCK_SH)
+	// LockEX places an exclusive lock.
+	// Only one process may hold an exclusive lock for a given file at a given time.
+	LockEX = int(C.LOCK_EX)
+	// LockUN removes and existing lock held by this process.
+	LockUN = int(C.LOCK_UN)
+	// LockNB can be ORed with any of the above to make a nonblocking call.
+	LockNB = int(C.LOCK_NB)
+)
+
+// Flock applies or removes an advisory lock on an open file.
+//
+// Implements:
+//  int ceph_flock(struct ceph_mount_info *cmount, int fd, int operation, uint64_t owner);
+func (f *File) Flock(operation int, owner uint64) error {
+	if err := f.validate(); err != nil {
+		return err
+	}
+
+	// validate the operation values before passing it on.
+	tempOperation := operation
+	tempOperation &^= LockNB
+	switch tempOperation {
```

Comment:
can you write `switch operation &^ LockNB` here, or does that not work?

Reply:
That works, made the change.